File: services/core-go/internal/middleware/requestid.go
package middleware

import "net/http"

func RequestID(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		id := r.Header.Get("X-Request-Id")
		if id != "" {
			w.Header().Set("X-Request-Id", id)
		}
		next.ServeHTTP(w, r)
	})
}

File: services/api-node/src/middleware/requestId.js
import crypto from 'crypto';
import { log } from '../logger.js';

export function requestId(req, res, next) {
  const id = req.headers['x-request-id'] || crypto.randomUUID();
  req.requestId = id;
  res.setHeader('x-request-id', id);
  log('info', 'requestId middleware', { requestId: id });
  next();
}
